backend/common/src/main/java/org/eclipse/sw360/cyclonedx/CycloneDxBOMImporter.java

```diff
@@ -987,16 +987,115 @@ public String getComponetNameById(String id, User user) throws SW360Exception {
-                String secondSegment = urlParts[2].replaceAll("\\.git.*", "").replaceAll("#.*", "");
-                vcs = "https://github.com/" + firstSegment + "/" + secondSegment;
-                return vcs;
-            } else {
-                log.error("Invalid GitHub repository URL: " + vcs);
+        if (vcs.contains("github.com")) {
+            vcs = "https://" + vcs.substring(vcs.indexOf("github.com")).trim();
+            try {
+                URI uri = URI.create(vcs);
+                String[] urlParts = uri.getPath().split("/");
+
+                if (urlParts.length >= 3) {
+                    String firstSegment = urlParts[1];
+                    String secondSegment = urlParts[2].replaceAll("\\.git.*", "").replaceAll("#.*", "");
+                    vcs = "https://github.com/" + firstSegment + "/" + secondSegment;
+                    return vcs;
+                } else {
+                    log.error("Invalid GitHub repository URL: {}", vcs);
+                }
+            } catch (IllegalArgumentException e) {
+                log.error("Invalid URL format: {}", vcs, e);
+            }
+        }else if(vcs.contains("gitlab.com")){
+            // GitLab repository URL Format: https://gitlab.com/supplier/name
+            vcs = "https://" + vcs.substring(vcs.indexOf("gitlab.com")).trim();
+            try{
+                URI uri = URI.create(vcs);
+                String[] urlParts = uri.getPath().split("/");
+                if (urlParts.length >= 3) {
+                    String firstSegment = urlParts[1];
+                    String secondSegment = urlParts[2].replaceAll("\\.git.*", "").replaceAll("#.*", "");
+                    vcs = "https://gitlab.com/" + firstSegment + "/" + secondSegment;
+                    return vcs;
+                } else {
+                    log.error("Invalid GitLab repository URL: " + vcs);
+                }
+            } catch (IllegalArgumentException e) {
+                log.error("Invalid URL format: {}", vcs, e);
+            }
+        }else if(vcs.contains("bitbucket.org")){
+            // Bitbucket repository URL Format: https://bitbucket.org/supplier/name
+            vcs = "https://" + vcs.substring(vcs.indexOf("bitbucket.org")).trim();
+            try{
+                URI uri = URI.create(vcs);
+                String[] urlParts = uri.getPath().split("/");
+                if (urlParts.length >= 3) {
+                    String firstSegment = urlParts[1];
+                    String secondSegment = urlParts[2].replaceAll("\\.git.*", "").replaceAll("#.*", "");
+                    vcs = "https://bitbucket.org/" + firstSegment + "/" + secondSegment;
+                    return vcs;
+                } else {
+                    log.error("Invalid Bitbucket repository URL: " + vcs);
+                }
+            } catch (IllegalArgumentException e) {
+                log.error("Invalid URL format: {}", vcs, e);
+            }
```

Comment:
There is lot of code repetition for github, gitlab and bitbucket. Can this be separated out as a function?